blob/service.go

```diff
@@ -0,0 +1,279 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	apptypes "github.com/celestiaorg/celestia-app/x/blob/types"
+	"github.com/celestiaorg/nmt/namespace"
+
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+var (
+	ErrBlobNotFound = errors.New("blob: not found")
+
+	log = logging.Logger("blob")
+)
+
+// TODO(@vgonkivs): remove after bumping celestia-app
+// defaultMinGasPrice is the default min gas price.
+const defaultMinGasPrice = 0.001
+
+// headerGetter fetches header by the provided height
+type headerGetter func(context.Context, uint64) (*header.ExtendedHeader, error)
```

Comment:
Yes, the idea is to pass `header/service.GetByHeight` that we merged yesterday

Reply:
can we just define this function on the struct to avoid another type definition? It's nit so i'm not gonna die on hill but i think it'd be nicer to just 

```
type Service struct {
	// accessor dials the given celestia-core endpoint to submit blobs.
	accessor *state.CoreAccessor
	// shareGetter retrieves the EDS to fetch all shares from the requested header.
	shareGetter  share.Getter
	// TODO
	headerGetter func(context.Context, uint64) (*header.ExtendedHeader, error)
}

```

Reply:
added it just for alias purposes.